dvc/repo/tag/list.py

```diff
@@ -1,3 +1,7 @@
+from dvc.repo import locked
+
+
+@locked
```

Comment:
It it supposed to reset it because it no longer guarantees that it will stay intact. But `locked` here is about locking the dvc repository, so that no other dvc instance can add/remove dvcfiles from the project, which would affect `self.collect` used below.

Reply:
Ok, thanks for the clarification, I only looked at `locked` as a mechanism caching the graph. It makes sense.